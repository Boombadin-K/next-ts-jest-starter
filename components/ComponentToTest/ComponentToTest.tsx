// Don't forget to import 'react'
import React from 'react';

// Export Props type/interface for better code suggestion on test file
export interface ComponentToTestProps {
  id?: string;
}

const ComponentToTest: React.FC<ComponentToTestProps> = ({ id }) => {
  return (
    <div id={id}>
      This is test component
    </div>
  );
};

export default ComponentToTest;
