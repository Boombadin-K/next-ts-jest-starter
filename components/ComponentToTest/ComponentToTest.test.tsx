/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import ComponentToTest, { ComponentToTestProps } from './ComponentToTest';

let componentToTest: ShallowWrapper;
const defaultProps: ComponentToTestProps = {
  id: 'jest-test',
};

describe('<ComponentToTest />', () => {
  beforeEach(() => {
    componentToTest = shallow(<ComponentToTest {...defaultProps} />);
  });

  it('should render correct HTML element', () => {
    expect(componentToTest).toMatchSnapshot();
    expect(componentToTest.find('div')).toHaveLength(1);
  });
});
