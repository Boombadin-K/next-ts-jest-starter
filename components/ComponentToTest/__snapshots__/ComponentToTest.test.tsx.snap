// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComponentToTest /> should render correct HTML element 1`] = `
<div
  id="jest-test"
>
  This is test component
</div>
`;
